watchdog-agent/src/main/java/io/github/algomaster99/Terminator.java

```diff
@@ -27,17 +24,21 @@ public byte[] transform(
                     Class<?> classBeingRedefined,
                     ProtectionDomain protectionDomain,
                     byte[] classfileBuffer) {
+
                 return isLoadedClassWhitelisted(className, classfileBuffer);
             }
         });
     }
 
     private static byte[] isLoadedClassWhitelisted(String className, byte[] classfileBuffer) {
         Map<String, List<Provenance>> fingerprints = options.getFingerprints();
-        if (RuntimeClass.isProxyClass(classfileBuffer)) {
+        if (RuntimeClass.isProxyClass(classfileBuffer)
+                || RuntimeClass.isGeneratedClassExtendingMagicAccessor(classfileBuffer)) {
             return classfileBuffer;
         }
-        if (INTERNAL_PACKAGES.stream().anyMatch(className::startsWith)) {
+        if (className.contains("$")) {
+            // FIXME: we need to check inner classes without loading them. Maybe add the hashes for inner classes in the
+            // fingerprints?
```

Comment:
I think it is added by default.
```
public class A {
static class B { }
}
```
will be compiled to `A.class` and `A$B.class`. Or do you mean something else?

Reply:
It will be compiled to a single class file. The files are currently missing in the index for jars.